### ISSUE TITLE
[no-noninteractive-element-interactions] should accept the dialog role

### ISSUE THREAD

**Issue by dfernandez79** [jsx-eslint/eslint-plugin-jsx-a11y#741]
Currently if you have code like this:

```jsx
<div role="dialog" onKeyPress={handleEscKey}>{contents}</div>
```

The rule `no-noninteractive-element-interactions` will fail. However, closing a dialog with the `Esc` is a common pattern also suggested by the WAI-ARIA practices site: https://www.w3.org/TR/wai-aria-practices-1.2/examples/dialog-modal/datepicker-dialog.html (the linked example has a keyDown handler in the `div` with the `dialog` role).

The lint rule passes with `role="presentation"` which is incorrect for this case.

**Comment by dfernandez79** [comment]
I'm closing this issue.

If the "dialog" contains any interactive elements the work around of moving the handler down will work.

I prefer to use bubbling to handle this, but also I'm not so happy about having a `onKeyPress` handler in an non-focusable element.
